cpp/include/raft/core/resource/user_resource.hpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <raft/core/resource/resource_types.hpp>
+#include <raft/core/resources.hpp>
+
+#include <memory>
+#include <typeindex>
+
+namespace raft::resource {
+
+class user_resource : public resource {
```

Comment:
Please make sure this gets added to the docs (and is well documented)

Reply:
Thanks for the reminder!

Reply:
I don't think this has been added to the docs yet, has it? (along with the one above?)

Reply:
Seems to be present in docs/source/cpp_api/core_resources.rst changeset?

Reply:
Got it, the docs are pulling are pulling it in directly. Thanks.